scripts/commands/checkLinks.ts

```diff
@@ -25,9 +25,39 @@ import remarkGfm from "remark-gfm";
 import yargs from "yargs/yargs";
 import { hideBin } from "yargs/helpers";
 
+const DOCS_GLOBS_TO_CHECK = [
+  "docs/build/*",
+  "docs/run/*",
+  "docs/start/*",
+  "docs/transpile/*",
+  "docs/verify/*",
+  "docs/support.mdx",
+  "docs/api/migration-guides/*",
+  "docs/api/qiskit/*",
+  "docs/api/qiskit-ibm-runtime/*",
+  "docs/api/qiskit-ibm-provider/*",
+  "!docs/**/*.json",
+];
```

Comment:
```suggestion
const DOCS_GLOBS_TO_CHECK = [
  "docs/**/*.{ipynb,md,mdx}",
  "!docs/api/*/[0-9]*/*", // Historical versions
  "!docs/api/qiskit/release-notes/*", // Qiskit release notes
];
```

I like the way you defined `DOCS_GLOBS_TO_CHECK` because we can remove all the conditionals we had inside the `processDocsFiles` function. Just one question: do you think it might be a problem to define the files like this if someday we add another folder and forget to update the script? Forgetting to ignore a file is detectable through CI, but forgetting to add one can be easily overlooked. 

Maybe we can use a more general pattern, similar to what we had before, and explicitly write what files we are ignoring.

In any case, I still like your definition, and I'm good with it if you think it's best.